src/data/types/recordid.ts

```diff
@@ -99,11 +103,21 @@ export function isValidIdPart(v: unknown): v is RecordIdValue {
 }
 
 export function escapeIdPart(id: RecordIdValue): string {
-	return id instanceof Uuid
-		? `u"${id}"`
-		: typeof id === "string"
-			? escapeIdent(id)
-			: typeof id === "bigint" || typeof id === "number"
-				? escapeNumber(id)
-				: toSurrealqlString(id);
+    if (id instanceof Uuid) {
+        return `u"${id}"`;
+    }
+    
+    if (typeof id === "string" && /^\d{15,}$/.test(id)) {
+        return `⟨${id}⟩`; // Wrap long numeric strings in angle brackets
+    }
```

Comment:
Every possible string which can be parsed as a number needs to be escaped no matter the length. The escapeIdent method takes care of that, however it's logic to check if a string is only numbers depends on `parseInt`. With longer numbers that probably fails with `NaN` cause the number-like string not to be escaped. Could you instead for the `isOnlyNumbers` method used by `escapeIdent` to use this regex, or perhaps a simple for-loop? Not sure which one will be more performant

Reply:
Did a little check, a regex will indeed be the fastest 🙏